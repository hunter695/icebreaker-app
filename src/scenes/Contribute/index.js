import React, { Component } from 'react'
import styled from 'styled-components'
import Switcher from './components/Switcher'
import Submission from './components/Submission'

const Main = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  :first-child {
    margin-bottom: 32px;
  }
`

export default class Contribute extends Component {
  state = {
    type: 'icebreaker',
    text: '',
  }

  switchType = () => {
    const type = this.state.type === 'icebreaker' ? 'pickup' : 'icebreaker'
    this.setState({
      type,
    })
  }

  updateText = (event) => {
    const text = event.target.value
    this.setState({
      text,
    })
  }

  render() {
    return (
      <Main>
        <Switcher
          name="contribution"
          choices={['icebreaker', 'pickup']}
          checked={this.state.type === 'pickup'}
          onChange={this.switchType}
        />
        <Submission
          type={this.state.type}
          text={this.state.text}
          onClick={() => console.log('click')}
          onChange={this.updateText}
        />
      </Main>
    )
  }
}
