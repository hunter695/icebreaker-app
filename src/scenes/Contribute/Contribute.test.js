import React from 'react'
import renderer from 'react-test-renderer'
import Contribute from './'

describe('<Contribute />', () => {
  it('renders UI', () => {
    const tree = renderer.create(<Contribute />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
