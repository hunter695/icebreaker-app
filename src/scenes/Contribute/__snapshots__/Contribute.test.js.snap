// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contribute /> renders UI 1`] = `
<main
  className="kekJLV"
>
  <div
    className="hwytNh"
  >
    <span
      className="iFFOKQ"
    >
      icebreaker
    </span>
    <div
      className="dZlebm"
    >
      <input
        id="contribution"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="contribution"
      />
    </div>
    <span
      className="dZcTbW"
    >
      pickup
    </span>
  </div>
  <div
    className="qYhrK"
  >
    <input
      className="gZvYYU"
      onChange={[Function]}
      placeholder="Enter your icebreaker"
      value=""
    />
    <button
      className="dheVMG"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</main>
`;
